### PR TITLE
Refactor test_epochs.py::test_split_saving (2 out of 2)

### LINKED ISSUE #11870: Inconsistent splits naming between `Raw` and `Epochs` when using `BIDS` schema
### Description of the problem

Not really a bug, more of an inconsistent behaviour.
Saving large `Raw` and `Epochs` data with `split_naming='bids'` uses different naming schema. 

When I save large `Raw`, I get suffixes `split-01` and `split-02`, etc. but when saving `Epochs`, the first split is saved without suffix, while the second one has `split-01`. See the code snippet below for details. All asserts pass for me.

Is there a reason for such difference?

### Steps to reproduce

```Python
import mne
from pathlib import Path

sample_data_folder = mne.datasets.sample.data_path()
sample_data_raw_file = sample_data_folder / "MEG" / "sample" / "sample_audvis_raw.fif"
raw = mne.io.read_raw_fif(sample_data_raw_file, verbose=False).crop(tmax=60)
ev = mne.find_events(raw)
epochs = mne.Epochs(raw, ev, tmin=-0.3, tmax=0.7)

raw.save("raw_meg.fif", split_naming="bids", split_size="50MB")
epochs.save("epochs_epo.fif", split_naming="bids", split_size="50MB")

# all these asserts pass
assert not Path("raw_meg.fif").is_file()
assert Path("raw_split-01_meg.fif").is_file()
assert Path("raw_split-02_meg.fif").is_file()

assert Path("epochs_epo.fif").is_file()  # this should fail though
assert Path("epochs_split-01_epo.fif").is_file()
assert not Path("epochs_split-02_epo.fif").is_file()  # this should fail as well
```

### Link to data

_No response_

### Expected results

These two asserts should fail
```python
assert Path("epochs_epo.fif").is_file()
assert not Path("epochs_split-02_epo.fif").is_file()
```

### Actual results

All asserts pass

### Additional information

Platform             macOS-13.4.1-arm64-arm-64bit
Python               3.11.3 (main, May 15 2023, 18:01:31) [Clang 14.0.6 ]
Executable           /Users/dmalt/Applications/miniconda3/envs/mne_cli_tools/bin/python3
CPU                  arm (8 cores)
Memory               Unavailable (requires "psutil" package)
Core
├☑ mne               1.4.2
├☑ numpy             1.25.2 (OpenBLAS 0.3.23.dev with 8 threads)
├☑ scipy             1.9.3
├☑ matplotlib        3.7.2Installed osx event loop hook.
 (backend=MacOSX)
├☑ pooch             1.7.0
└☑ jinja2            3.1.2

Numerical (optional)
├☑ sklearn           1.3.0
├☑ pandas            1.5.3
└☐ unavailable       numba, nibabel, nilearn, dipy, openmeeg, cupy

Visualization (optional)
└☐ unavailable       pyvista, pyvistaqt, ipyvtklink, vtk, qtpy, ipympl, pyqtgraph, mne-qt-browser

Ecosystem (optional)
└☐ unavailable       mne-bids, mne-nirs, mne-features, mne-connectivity, mne-icalabel, mne-bids-pipeline

### PR DESCRIPTION
**Pull request by dmalt**
Continuation of #11880

Leg work to fix #11870.
- Make adding new test cases more straightforward via parametrisation
- Fix problems with existing tests
- Add test cases to stress problems with existing implementation

### CODE REVIEW

**Comment by dmalt** [comment]
Ready for review as soon as the tests pass.

File: mne/tests/test_epochs.py [hunk 1]
```diff
@@ -1477,17 +1477,20 @@ def test_epochs_io_preload(tmp_path, preload):
 
 @pytest.fixture(scope="session")
 def epochs_factory():
-    """Function to create fake Epochs object."""  # noqa: D401 (imperative mood)
+    """Function to create fake Epochs object.
+
+    Metadata and concat address gh-5102, gh-7897.
+
+    """  # noqa: D401 (imperative mood)
```
Comment (larsoner) [inline mne/tests/test_epochs.py:1484]:
```suggestion
    """Create fake Epochs object.

    Metadata and concat address gh-5102, gh-7897.
    """
```
The linter is right to complain and it's less work (and code) to make it happy than it is to ignore it here

**Review by larsoner — APPROVED** [review]
Just one minor gripe which I'll commit, which will also get all CIs to run.

FYI for CircleCI can you try going to the site and logging in with GitHub? It should make it happy to build the docs for you

**Comment by dmalt** [comment]
> Just one minor gripe which I'll commit, which will also get all CIs to run.

Great, thanks!

> FYI for CircleCI can you try going to the site and logging in with GitHub? It should make it happy to build the docs for you

Looks like it worked.

**Comment by dmalt** [comment]
@larsoner looks like `ci/circleci: pytest-macos-arm64` is stuck. Can we merge it before `main` diverges, so I can start working on the fix?

**Comment by larsoner** [comment]
Thanks @dmalt !

**Pull request merged**